pkg/flow/grpc-secrets.go

```diff
@@ -2,227 +2,129 @@ package flow
+			if err != nil {
+				return err
+			}
+			time.Sleep(time.Second * 5)
 		}
 	}
-	phash = nhash
-
-	more := sub.Wait(ctx)
-	if !more {
-		return nil
-	}
-
-	goto resend
 }
 
 func (flow *flow) SearchSecret(ctx context.Context, req *grpc.SearchSecretRequest) (*grpc.SearchSecretResponse, error) {
 	flow.sugar.Debugf("Handling gRPC request: %s", this())
 
-	cached := new(database.CacheData)
-
-	err := flow.database.NamespaceByName(ctx, cached, req.GetNamespace())
+	ns, err := flow.edb.NamespaceByName(ctx, req.GetNamespace())
 	if err != nil {
 		return nil, err
 	}
-
-	p, err := getPagination(req.Pagination)
+	_, store, _, rollback, err := flow.beginSqlTx(ctx)
 	if err != nil {
 		return nil, err
 	}
+	defer rollback()
 
-	namespace := cached.Namespace.ID.String()
-	name := req.GetKey()
-
-	request := &secretsgrpc.SearchSecretRequest{
-		Namespace: &namespace,
-		Name:      &name,
-	}
-
-	response, err := flow.secrets.client.SearchSecret(ctx, request)
-	if err != nil {
-		return nil, err
-	}
-
-	cpds := newCustomPaginationDataSecrets()
-	pagination := newCustomPagination(cpds)
-	for i := range response.Secrets {
-		cpds.Add(response.Secrets[i].GetName())
-	}
-
-	cx, err := pagination.Paginate(p)
+	list, err := store.Secrets().Search(ctx, ns.ID, req.GetKey())
 	if err != nil {
 		return nil, err
 	}
 
 	var resp grpc.SearchSecretResponse
 
-	resp.Namespace = cached.Namespace.Name
+	resp.Namespace = ns.Name
 	resp.Secrets = new(grpc.Secrets)
-	resp.Secrets.PageInfo = new(grpc.PageInfo)
+	resp.Secrets.PageInfo = nil
 
-	err = bytedata.ConvertDataForOutput(cx, &resp.Secrets)
-	if err != nil {
-		return nil, err
-	}
+	resp.Secrets.Results = bytedata.ConvertSecretsToGrpcSecretList(list)
 
 	return &resp, nil
 }
 
 func (flow *flow) SetSecret(ctx context.Context, req *grpc.SetSecretRequest) (*grpc.SetSecretResponse, error) {
 	flow.sugar.Debugf("Handling gRPC request: %s", this())
 
-	cached := new(database.CacheData)
-
-	err := flow.database.NamespaceByName(ctx, cached, req.GetNamespace())
+	ns, err := flow.edb.NamespaceByName(ctx, req.GetNamespace())
 	if err != nil {
 		return nil, err
 	}
-
-	namespace := cached.Namespace.ID.String()
-	name := req.GetKey()
-
-	request := &secretsgrpc.SecretsStoreRequest{
-		Namespace: &namespace,
-		Name:      &name,
-		Data:      req.GetData(),
+	_, store, commit, rollback, err := flow.beginSqlTx(ctx)
+	if err != nil {
+		return nil, err
 	}
+	defer rollback()
 
-	_, err = flow.secrets.client.StoreSecret(ctx, request)
+	err = store.Secrets().Set(ctx, &core.Secret{
+		ID:          uuid.New(),
+		NamespaceID: ns.ID,
+		Name:        req.GetKey(),
+		Data:        req.GetData(),
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	flow.logger.Infof(ctx, cached.Namespace.ID, cached.GetAttributes(recipient.Namespace), "Created namespace secret '%s'.", req.GetKey())
-	flow.pubsub.NotifyNamespaceSecrets(cached.Namespace)
+	if err = commit(ctx); err != nil {
+		return nil, err
+	}
+	// TODO: Alax, please look into this.
```

Comment:
This typo makes me unsure if you're asking for me or for Alexander. Can you elaborate what the concern is here?

Reply:
If it's just the commented out stuff, if you insert a TODO here I think it's safe to merge without for now.